alacritty_terminal/src/grid/mod.rs

```diff
@@ -188,25 +187,27 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
     pub fn update_history(&mut self, history_size: usize, template: &T) {
         self.raw.update_history(history_size, Row::new(self.cols, &template));
         self.max_scroll_limit = history_size;
-        self.scroll_limit = min(self.scroll_limit, history_size);
-        self.display_offset = min(self.display_offset, self.scroll_limit);
+        if history_size < self.history_size() {
+            self.decrease_scroll_limit(self.history_size() - history_size);
+        }
```

Comment:
Does it make sense to call `decrease_scroll_limit` here, or should all changes to the raw buffer be made in `update_history`?

Reply:
I was following the old code, the old code was updating scroll_limit to a new value, so it was decreasing it, so this call makes sense from the point of the old code.

Reply:
Well if it is **required** that the raw buffer is modified when the history is shrunk, it might make sense to enforce that in the `storage.rs` file with the `update_history` method.

My only question would be if it actually is possible to do things there, or if it lacks some information that only the grid can provide to `shrink_lines`.

But having to call two separate methods on `grid.raw` to perform one operation is usually a bad thing and will cause bugs in the usage of the API.

Reply:
storage.rs doesn't know what history is, it's just a ring buffer, so let's just keep logic that doesn't make sense to it in grid.

Reply:
Of course `storage.rs` knows what history is, that's why we have a field called `visible_lines` on `Storage`. We always know that `len - visible_lines` is the history.

We're already calling `shrink_lines(old_history - new_history)` inside `update_history` in `storage.rs`. So if it turns out that `history_size < self.history_size()` after that, there's a bug somewhere in the `update_history` function.

This code should not be necessary and removing it shouldn't do anything.